components/authz-service/engine/opa/opa_internal_test.go

```diff
@@ -549,3 +429,864 @@ func randomPolicies(i int) []map[string]interface{} {
+	l, err := logger.NewLogger("text", "debug")
+	require.NoError(b, err, "init logger")
+	s, err := New(ctx, l)
+	require.NoError(b, err, "init state")
+
+	policyCount := 20 // keep this constant while increasing roleCount
+	roleCount := []int{0, 5, 10, 20, 50, 100}
+
+	for _, count := range roleCount {
+		policies, roles := v2BaselineAndRandomPoliciesAndRoles(policyCount, count)
+		s.v2p1Store = inmem.NewFromObject(map[string]interface{}{
+			"policies": policies,
+			"roles":    roles,
+		})
+
+		err = s.makeAuthorizedProjectPreparedQuery(ctx)
+		require.NoError(b, err, "prepared authorized project query")
+
+		b.Run(fmt.Sprintf("store with %d custom policies and %d custom roles", policyCount, count), func(b *testing.B) {
+			var resp []string
+			var err error
+			for n := 0; n < b.N; n++ {
+				resp, err = s.V2ProjectsAuthorized(ctx, []string{"user:local:test"}, "compliance:profiles:list", "compliance:profiles", allProjects)
+				if err != nil {
+					b.Error(err)
+				}
+			}
+			projectsResponse = resp
+		})
+	}
+}
+
+func BenchmarkV2FilterAuthorizedProjectsWithIncreasingRoles(b *testing.B) {
+	ctx := context.Background()
+
+	l, err := logger.NewLogger("text", "debug")
+	require.NoError(b, err, "init logger")
+	s, err := New(ctx, l)
+	require.NoError(b, err, "init state")
+
+	policyCount := 20 // keep this constant while increasing roleCount
+	roleCount := []int{0, 5, 10, 20, 50, 100}
+
+	for _, count := range roleCount {
+		policies, roles := v2BaselineAndRandomPoliciesAndRoles(policyCount, count)
+		s.v2p1Store = inmem.NewFromObject(map[string]interface{}{
+			"policies": policies,
+			"roles":    roles,
+		})
+
+		err = s.makeAuthorizedProjectPreparedQuery(ctx)
+		require.NoError(b, err, "prepared authorized project query")
+
+		b.Run(fmt.Sprintf("store with %d custom policies and %d custom roles", policyCount, count), func(b *testing.B) {
+			var resp []string
+			var err error
+			for n := 0; n < b.N; n++ {
+				resp, err = s.V2FilterAuthorizedProjects(ctx, []string{"user:local:test"})
+				if err != nil {
+					b.Error(err)
+				}
+			}
+			projectsResponse = resp
+		})
+	}
+}
+
+func BenchmarkV2ProjectsAuthorizedWithIncreasingProjects(b *testing.B) {
+	ctx := context.Background()
+
+	l, err := logger.NewLogger("text", "debug")
+	require.NoError(b, err, "init logger")
+	s, err := New(ctx, l)
+	require.NoError(b, err, "init state")
+
+	projectCounts := []int{5, 20, 100, 200, 300}
+	member := "user:local:test"
+
+	for _, projCount := range projectCounts {
+		policyMap, projectIDs := v2BaselineAndProjectPolicies(projCount)
+
+		_, roleMap := v2BaselinePoliciesAndRoles()
+
+		s.v2p1Store = inmem.NewFromObject(map[string]interface{}{
+			"policies": policyMap,
+			"roles":    roleMap,
+		})
+
+		err = s.makeAuthorizedProjectPreparedQuery(ctx)
+		require.NoError(b, err, "prepared authorized project query")
+
+		b.Run(fmt.Sprintf("store with %d projects %d policies, and %d roles", projCount, len(policyMap), len(roleMap)), func(b *testing.B) {
```

Comment:
```suggestion
		b.Run(fmt.Sprintf("store with %d projects, %d policies, and %d roles", projCount, len(policyMap), len(roleMap)), func(b *testing.B) {
```
Same fix needed in other spots, too.